renderer/brave_url_loader_throttle_provider_impl.cc

```diff
@@ -78,5 +88,25 @@ BraveURLLoaderThrottleProviderImpl::CreateThrottles(
     throttles.emplace_back(std::move(onion_domain_throttle));
   }
 #endif
+  // AI Chat
+#if BUILDFLAG(ENABLE_AI_CHAT)
+  if (ai_chat::features::IsAIChatEnabled() && local_frame_token.has_value() &&
+      content::RenderThread::IsMainThread()) {
+    content::RenderFrame* render_frame = content::RenderFrame::FromWebFrame(
+        blink::WebLocalFrame::FromFrameToken(local_frame_token.value()));
+    auto* page_content_delegate =
+        ai_chat::PageContentExtractor::Get(render_frame);
+    if (page_content_delegate) {
+      std::unique_ptr<ai_chat::AIChatResourceSnifferThrottle>
+          ai_chat_resource_throttle =
+              ai_chat::AIChatResourceSnifferThrottle::MaybeCreateThrottleFor(
+                  page_content_delegate->GetWeakPtr(), request.url,
+                  base::SingleThreadTaskRunner::GetCurrentDefault());
```

Comment:
why it is not just `SequencedTaskRunner`?

Reply:
That's what SpeedreaderThrottle had, but I suppose it has different requirements

Reply:
yes I think in the speedreader case we were worrying about running rust code on different threads